### ISSUE TITLE
Unsound TypedDict updates are allowed

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ty#4558]
### Summary

Ty allows this:

```python
from typing import TypedDict


class Source(TypedDict):
    name: str


class SourceWithBadCount(Source):
    count: str


class Destination(TypedDict):
    name: str
    count: int


def merge(destination: Destination, source: Source) -> None:
    destination.update(source)  # mypy --strict flags this; ty accepts it


def capybara() -> None:
    destination: Destination = {"name": "before", "count": 0}
    source: SourceWithBadCount = {"name": "after", "count": "not an int"}

    merge(destination, source)

    print(destination["count"] + 1)  # TypeError: count is now a str
```

But it fails at runtime. For soundness, `.update` on a TypedDict should be rejected if the dict being updated in may contain incompatible keys.

Among other type checkers, pyrefly and pyright catch this; mypy catches this only in strict mode (it's part of `--extra-checks`); zuban and pycroscope don't catch it. The spec is not precise about what's required here. Ty could also consider making this check optional, but I do think strict mode should catch it.

### Version

0.0.82

**Comment by carljm** [comment]
Thanks! I feel like this should be caught and should be on by default, not an opt-in strict diagnostic.
